### ISSUE TITLE
Extract more information from duration/date/filesize objects

### ISSUE THREAD

**Issue by bew** [nushell/nushell#14469]
### Related problem

Hello o/

I like to use nushell for date/duration/filesize conversions, but there is something that I need from time to time that seems to be unavailable at the moment:
- converting to a subset of the value, like `hours to milliseconds` to get the total number of milliseconds in a number of hours (~ `1hr | get total_ms` => `3_600_000ms`)
- getting a superset of the value, like `seconds to hours` to get the number of hours in a (potentially large) number of seconds (~ `123456sec | get total_hours` => `34hr`)

These example could be useful for duration but could also be useful for `filesize` or `date` objects 🤔

I've found that we can do `123456sec | into record` to get as data the information in the humanized form `1day 10hr 17min 36sec` but even here we are missing some fields like `milliseconds` or `weeks` that could be needed when manipulating unknown values.

### Describe the solution you'd like

Something like this:
`1wk | into record | get total_hours`
`(date now) - 2019-05-01 | into record | get total_days`
`12sec | into record | get hours` (0)

### Describe alternatives you've considered

I could handle all edge cases, do sums myself, but it feels like a huge pain for seemingly simple operations 🤔

**Comment by NotTheDr01ds** [comment]
What about something like:

```nushell
def duration-units [unit: string] {
  format duration $unit | split row ' ' | { $unit: ($in.0 | into float) }
}

1min | duration-units hr
# => ╭────┬──────╮
# => │ hr │ 0.02 │
# => ╰────┴──────╯

(date now) - 2019-05-01 | duration-units day | get day
# => 2039.04
```

**Comment by fdncred** [comment]
```nushell
❯ 1hr | format duration ms
3600000 ms
❯ 123456sec | format duration hr
34.29 hr
```

**Comment by NotTheDr01ds** [comment]
@fdncred Right, but that returns a string, with the unit, and I assume from the original question that the goal would be to treat it as a number.  That's why I went ahead and converted it to its individual parts in the command above.

**Comment by bew** [comment]
Ah interesting!
https://www.nushell.sh/commands/docs/format.html
And it exists for duration and filesize.
It also exists for date but seems limited to extracting specific values, not transforming to specific unit 🤔

The string-to-int conversion feels like a hack though, there's no guarantee that this `split row ' '` is gonna hold long-term 

The unit in OP was to make it clearer, but yes usually it's to get the number

**Comment by LoicRiegel** [comment]
What about this solution:
```nu
# duration to ns
> 10sec | into int
# => 10_000_000_000

# duration to other units with a --unit flag
> 10sec | into int --unit ms
# => 10_000
```

We'd have to make the new `--unit` flag also work with filesize units

**Comment by Bahex** [comment]
@LoicRiegel I don't think this is necessary, since it's possible to divide durations. 
```nushell
10sec / 1ms
# => 10_000
```

**Comment by LoicRiegel** [comment]
True @Bahex , I hadn't even thought of that! Maybe we should then just add an example to demonstrate this?

Also, maybe ``into int --unit`` could make life a little bit easier if you got values in tables or records I guess, but maybe that's not worth it
```nu
~> [1hr 2day] | into int --unit ms

# would be a bit simpler than

~> [1hr 2day] | each {|it| $it / (1ms) }
╭───┬──────────────╮
│ 0 │   3600000.00 │
│ 1 │ 172800000.00 │
╰───┴──────────────╯
```
BTW I have no idea why I'm gettings floats as a result!

**Comment by fdncred** [comment]
ya, I'm fine with having `[1hr 2day] | into int --unit ms`. I think it's worth it.

**Comment by LoicRiegel** [comment]
Everyone aggress? @Bahex @NotTheDr01ds 
I prefer to ask before implementing :)

**Comment by Bahex** [comment]
`into int` also works with `filesize`, if we add `--unit` for `duration`, we should support the same for `filesize`.

That's why I consider the division more intuitive and "the nu way", it works with all relevant types.
```nushell
1GiB / 1MiB
# => 1024

10sec / 1ms
# => 10_000
```

Actually, I'm not sure if we really needed `into duration --unit` either.
```nushell
1_234 | into duration --unit ms
# => 1sec 234ms

# VS

1_234 * 1ms
# => 1sec 234ms
```

**Comment by LoicRiegel** [comment]
> into int also works with filesize, if we add --unit for duration, we should support the same for filesize

Yes absolutely

> Actually, I'm not sure if we really needed into duration --unit either.

Hmm interesting

I don't know, both solutions seem good, I don't have a clear preference... But if we decide to not implement ``into int --unit`` and let users do divisions, I would think that removing the ``--unit`` from ``into datetime`` makes sense. And we could just add two additional examples in the help to guide users

**Comment by LoicRiegel** [comment]
Still no consensus? @fdncred @Bahex

**Comment by fdncred** [comment]
> ya, I'm fine with having `[1hr 2day] | into int --unit ms`. I think it's worth it.

I think we should do it like this. This way both ways are possible, bahex's way with math and the way with the command.

**Comment by LoicRiegel** [comment]
Okay, I'm on it
